src/client/api/event.c

```diff
@@ -1020,7 +1071,11 @@ daos_event_init(struct daos_event *ev, daos_handle_t eqh,
 		daos_eq_putref(eqx);
 	} else {
 		evx->evx_ctx = daos_eq_ctx;
-		evx->evx_sched = &daos_sched_g;
+		evx->evx_sched = daos_sched_get();
```

Comment:
Then per daos api will create a new scheduler ? It looks like you can move tse_task_schedule(io_task) outside of loop in dc_array_to() fix the issue? No?

Reply:
> Then per daos api will create a new scheduler ? It looks like you can move tse_task_schedule(io_task) outside of loop in dc_array_to() fix the issue? No?

For the case 1 mentioned in the commit msg, yes moving "tse_task_schedule(io_task, false);" outside while() loop should be able to fix the issue.

But there are quite some other cases, need careful check/change. For example dfs_open issue in case 2, and some other strange problem found by running PR1235.